### ISSUE TITLE
NullPointerError for GET requests with an empty body

### ISSUE THREAD

**Issue by cbeer** [jayway/rest-assured#642]
We're seeing a NullPointerError for GET requests that have an empty body:

```
java.lang.NullPointerException
    at com.jayway.restassured.internal.support.CloseHTTPClientConnectionInputStreamWrapper.read(CloseHTTPClientConnectionInputStreamWrapper.java:49)
    at java.io.FilterInputStream.read(FilterInputStream.java:133)
    at sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:283)
    at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:325)
    at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:177)
    at java.io.InputStreamReader.read(InputStreamReader.java:184)
    at java.io.Reader.read(Reader.java:140)
```

Here's the line in our code where things break:

https://github.com/w3c/ldp-testsuite/blob/master/src/main/java/org/w3/ldp/testsuite/mapper/RdfObjectMapper.java#L48

Tracing my way through rest-assured, it looks like the NPE error is initially caused here:

https://github.com/jayway/rest-assured/blob/master/rest-assured/src/main/groovy/com/jayway/restassured/internal/RequestSpecificationImpl.groovy#L1969

And propagates through to:
https://github.com/jayway/rest-assured/blob/master/rest-assured/src/main/groovy/com/jayway/restassured/internal/RestAssuredResponseOptionsGroovyImpl.groovy#L182-L188

**Comment by johanhaleby** [comment]
Thanks for pointing out. Do you think it will help if we just replace null with empty string?

**Comment by cbeer** [comment]
Makes sense to me, and that would match the behavior of `asString`  and `asByteArray`.

**Comment by johanhaleby** [comment]
I've fixed this in master now. Thanks for a very detailed issue report. I've published a new snapshot: `2.8.1-SNAPSHOT`. Please try it out after having added the following repo:

```xml
<repositories>
        <repository>
            <id>sonatype</id>
            <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
            <snapshots />
        </repository>
</repositories>
```

**Comment by cbeer** [comment]
:+1: thanks, the snapshot release fixes our problem.
